CASSANDRA-17499 Remove global Guardrails Enable flag

thingtwin1 opened the pull request.

Commit:
squashed removed global enable commit

jmckenzie-dev commented on src/java/org/apache/cassandra/cql3/statements/schema/CreateTableStatement.java:
```diff
@@ -125,10 +125,6 @@ public void validate(ClientState state)
-        // optimization for the case where they are disabled, so we don't have to do the same check on every guardrail
-        if (Guardrails.enabled(state))
-        {
             // Guardrail on table properties
```
Fix indentation on this block now that you've removed the Guardrails.enabled check

jmckenzie-dev reviewed (approved):
Also have unused import in Guardrails.java (ClientState): https://github.com/apache/cassandra/pull/1533/files#diff-ee9b238deb86b4f297d2bd8a400cdfd799e33d08238bde9652b73a2f64a89c08R28

Probably want to get @adelapena to review as well but LGTM w/nits.

adelapena commented:
@thingtwin1 this patch seems based on a version of trunk that is more that one month old, and it's missing the 5 new guardrails that have been added since then.

thingtwin1 commented:
> @thingtwin1 this patch seems based on a version of trunk that is more that one month old, and it's missing the 5 new guardrails that have been added since then.

sorry about that, let me fix it now!

adelapena commented on src/java/org/apache/cassandra/db/guardrails/GuardrailsConfig.java:
```diff
@@ -46,13 +46,6 @@
-     *
-     * @return {@code true} if guardrails are enabled, {@code false} otherwise
-     */
-    boolean getEnabled();
```
There is a comment mentioning the global enable/disable flag it the class' JavaDoc, [here](https://github.com/apache/cassandra/blob/26861b173f172b663a86e03a835ac03038be13d1/src/java/org/apache/cassandra/db/guardrails/GuardrailsConfig.java#L33-L34), that should be updated. It could be updated to something like `This contains individual settings to control each guardrail.`. That same comment also exists on [`GuardrailsOptions`](https://github.com/apache/cassandra/blob/26861b173f172b663a86e03a835ac03038be13d1/src/java/org/apache/cassandra/config/GuardrailsOptions.java#L45-L46).

adelapena commented:
@thingtwin1 thanks for pulling those changes. I think it might be more practical to rebase on top of trunk with `git rebase trunk` instead of merging from trunk on top of the patch branch. That way the PR would show the proposed changes on top of trunk's commits, without the additional merge commit.

It seems that the removal of the property breaks the dtests [`GuardrailCollectionSizeOnSSTableWriteTest`](https://github.com/apache/cassandra/blob/935bcf1e8732a4138c15205896945c2f02ddb844/test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailCollectionSizeOnSSTableWriteTest.java#L57) and [`GuardrailItemsPerCollectionOnSSTableWriteTest`](https://github.com/apache/cassandra/blob/935bcf1e8732a4138c15205896945c2f02ddb844/test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailItemsPerCollectionOnSSTableWriteTest.java#L52). Those tests are still mentioning the removed property.

Commit:
fixed failing collection test

thingtwin1 commented:
> @thingtwin1 thanks for pulling those changes. I think it might be more practical to rebase on top of trunk with `git rebase trunk` instead of merging from trunk on top of the patch branch. That way the PR would show the proposed changes on top of trunk's commits, without the additional merge commit.
> 
> It seems that the removal of the property breaks the dtests [`GuardrailCollectionSizeOnSSTableWriteTest`](https://github.com/apache/cassandra/blob/935bcf1e8732a4138c15205896945c2f02ddb844/test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailCollectionSizeOnSSTableWriteTest.java#L57) and [`GuardrailItemsPerCollectionOnSSTableWriteTest`](https://github.com/apache/cassandra/blob/935bcf1e8732a4138c15205896945c2f02ddb844/test/distributed/org/apache/cassandra/distributed/test/guardrails/GuardrailItemsPerCollectionOnSSTableWriteTest.java#L52). Those tests are still mentioning the removed property.

rebased and fixed the failing tests

adelapena commented on src/java/org/apache/cassandra/db/guardrails/Guardrails.java:
```diff
@@ -244,29 +243,6 @@ private Guardrails()
-     */
-    public static boolean enabled(ClientState state)
-    {
-        return DatabaseDescriptor.isDaemonInitialized() && CONFIG_PROVIDER.getOrCreate(state).getEnabled();
```
We still need the call to `DatabaseDescriptor.isDaemonInitialized` so guardrails are not checked during startup nor by standalone tools like `scrub`. This call should be part of `Guardrail#enabled(ClientState)`.

Commit:
fixed pr requested changes

Commit:
trigger ci
